Add doc for recent target extension features (#6877)
numba/numba: docs/source/developer/target_extension.rst

> =========================
> Notes on Target Extension
> =========================

Reviewer: ```suggestion
==========================
Notes on Target Extensions
==========================
```

> .. warning:: All features and API described in this page are in-development and
>              may change without deprecation notice.

Reviewer: ```suggestion
.. warning:: All features and APIs described in this page are in-development and
             may change at any time without deprecation notices being issued.
```

Inheriting compiler flags from the caller
=========================================

Compiler flags, i.e. options such as ``fastmath``, ``nrt`` in
``@jit(nrt=True, fastmath=True))`` are specified per-function but their
effects are not well-defined---some flags affect the entire callgraph, some

> flags affect only the current function. Sometimes, it is necessary for callees

Reviewer: ```suggestion
flags affect only the current function. Sometimes it is necessary for callees
```

to inherit flags from the caller; for example the ``fastmath`` flag should be
infectious.

> To address the problem, the followings are needed:

Reviewer: ```suggestion
To address the problem, the following are needed:
```

> 1. Better define the semantic for compiler flags. Preferably, all flags should

Reviewer: ```suggestion
1. Better definitions for the semantics of compiler flags. Preferably, all flags should
```
